provider/types.go

```diff
@@ -0,0 +1,416 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"maps"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+// extraTypes augment the types defined by the upstream AWS provider
+var extraTypes = map[string]schema.ComplexTypeSpec{
```

Comment:
I moved this to its own file to make it easier to work with and add additional extraTypes sources (like the overlays)

Reply:
Thank you so much, I dread working in the super long resources.go.